install.sh

```diff
@@ -463,6 +463,51 @@ assert_usable_kernel() {
   fi
 }
 
+maybe_enable_userns_sysctl() {
+  # This function enables the Debian/Ubuntu-specific unprivileged
+  # userns sysctl, if the system has it and we want it.
+
+  if [ "$USE_DEFAULTS" != "yes" ] ; then
+    # Only do this when -d is passed. -d means "use defaults suitable for app development", and
+    # we want userns enabled for app development if possible since it enables UID randomization
+    # which helps catch app bugs. For the rest of the world, we're fine using the privileged
+    # sandbox instead.
+    return
+  fi
+
+  if [ "no" = "$CURRENTLY_UID_ZERO" ] ; then
+    # Not root. Can't do anything about it.
+    return
+  fi
+
+  if [ ! -e /proc/sys/kernel/unprivileged_userns_clone ]; then
+    # No such sysctl on this system.
+    return
+  fi
+
+  if [ "$(sysctl -n kernel.unprivileged_userns_clone)" = "1" ]; then
+    # Already enabled.
+    return
+  fi
+
+  # Enable it.
+  echo "NOTE: Enabling unprivileged user namespaces because you passed -d."
+  sysctl -wq kernel.unprivileged_userns_clone=1
```

Comment:
I'm concerned that if this fails, we have no code to handle the failure. The current install.sh does things this way:

```
    sysctl -w kernel.unprivileged_userns_clone=1 >/dev/null \
      || fail "E_SYSCTL_WRITING_FAILED" "'sysctl -w" \
              "kernel.unprivileged_userns_clone=1' failed. If you are inside docker, please run the" \
              "command manually inside your host and update /etc/sysctl.conf."
```

If the `sysctl` command fails, arguably we want to abort this function and continue with the install.

Same with the writing to `$SYSCTL_FILENAME`.

Curious if you agree.